events/tutorials/tutorial_import_user_events_gcs.md

```diff
@@ -0,0 +1,155 @@
+```bash
+pip install google-cloud-retail
+```
+
+**Tip**: Click the copy button on the side of the code box to paste the command in the Cloud Shell terminal to
+run it.
+
+## Set the PROJECT_NUMBER environment variable
+
+As you are going to run the code samples in your own Cloud Project, you should specify the **project_id** as an environment variable, it will be used in every request to the Retail API.
+
+You can find the ```project_number``` in the **Home/Dashboard/Project Info card**.
+
+Set the environment variable with a following command:
+```bash
+export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+```
+
+## Import user events from the Cloud Storage source
+
+To upload catalog data to the Cloud Storage bucket, create one or more JSON files with user events that do not exceed 2 GB each. You can set up to 100 JSON files in a single import request. For more information, refer to the [example of the user event in JSON format](https://cloud.google.com/retail/docs/user-events#formats)
+
+To check the example of an import user events request, open **events/import_user_events_gcs.py**.
+
+The field **```parent```** contains a **catalog name** along with a branch number you are going to import your
+user events to.
+
+The field **```input_config```** defines the **GcsSource** as an import source.
+
+To perform the user events import, open terminal and run the command:
+
+```bash
+python events/import_user_events_gcs.py
+```
+
+## Response analysis
+
+Once you have called the import user events method from the Retail API, the **import operation** has started.
+
+Importing may take some time depending on the size of user events set in your Cloud Source.
+
+The operation is completed when the field **```operation.done()```** is set to true. 
+
+Check the result, one of the following fields should be present:
+ - **```error```**, if the operation failed.
+ - **```result```**, if the operation was successful.
+
+You have imported valid user event objects into the catalog.
+
+Check the ```gcs_operation.metadata.success_count``` field to get the total number of the successfully imported events.
+
+The number of failures during the import is returned to the ```gcs_operation.metadata.failure_count``` field.
+
+The operation is successful, and the operation object contains a **```result```** field.
+Check it printed out in a Terminal. It should look like this: 
+
+```
+errors_config {
+  gcs_prefix: "gs://import_user_events/error"
+}
+import_summary {
+  joined_events_count: 13500
+}
+```
+
+## Errors appeared during the importing
+
+Now, let's try to import a few invalid user event objects and check the error message in the operation response. Note that in this case the operation itself is considered successful.
+
+The ```type``` and ```visitor_id``` fields are required, so if you remove them, you get the invalid user event objects. 
+
+There is a **```import_user_events_invalid.json```** file in the Cloud Storage bucket containing such an invalid user events.
```

Comment:
There is an **```import_user_events_invalid.json```** file in the Cloud Storage bucket containing such invalid user events.